sympy/polys/agca/extensions.py

```diff
@@ -67,55 +73,94 @@ def __rsub__(f, g):
     def __mul__(f, g):
         rep = f._get_rep(g)
         if rep is not None:
-            return ExtElem((f.rep*rep) % f.ext.mod, f.ext)
+            return ExtElem((f.rep * rep) % f.ext.mod, f.ext)
         else:
             return NotImplemented
 
     __rmul__ = __mul__
 
+    def _divcheck(f):
+        """Raise if division is not implemented for this divisor"""
+        if not f:
+            raise NotInvertible('Zero divisor')
+        elif f.ext.is_Field:
+            return True
+        elif f.rep.is_ground and f.ext.domain.is_unit(f.rep.rep[0]):
+            return True
+        else:
+            # Some cases like (2*x + 2)/2 over ZZ will fail here. It is
+            # unclear how to implement division in general if the ground
+            # domain is not a field so for now it was decided to restrict the
+            # implementation to division by invertible constants.
+            msg = (f"Can not invert {f} in {f.ext}. "
+                    "Only division by invertible constants is implemented.")
+            raise NotImplementedError(msg)
+
     def inverse(f):
         """Multiplicative inverse.
 
         Raises
         ======
+
         NotInvertible
             If the element is a zero divisor.
 
         """
-        if not f.ext.domain.is_Field:
-            raise NotImplementedError("base field expected")
-        return ExtElem(f.rep.invert(f.ext.mod), f.ext)
+        f._divcheck()
```

Comment:
Is there no need to check the return value.

Reply:
The `_divcheck` function raises an exception unless the element can be inverted.